lib/spack/spack/build_systems/cmake.py

```diff
@@ -90,10 +91,19 @@ class CMakePackage(PackageBase):
     #: for more information.
     generator = 'Unix Makefiles'
 
+    # cmake build types
+    build_types = ('Debug', 'Release', 'RelWithDebInfo', 'MinSizeRel')
+
     # https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html
     variant('build_type', default='RelWithDebInfo',
             description='CMake build type',
-            values=('Debug', 'Release', 'RelWithDebInfo', 'MinSizeRel'))
+            values=build_types)
+
+    # cmake conflicts with any build_type except Debug
+    for sbt in spack.build_types.debug_types:
+        for bt in build_types:
+            if bt != 'Debug':
+                conflicts('build_type=%s' % bt, when='spack_build_type=%s' % sbt)
```

Comment:
What is the final purpose of this PR? :thinking:

Reply:
To add this global ability to ask for something to be built with debug (and eventually other themes) - that’s the spack build type.

Reply:
And its for the BUILD SI project so we can make sure to build with debug symbols, which we typically need to extract symbols and metadata about the binary.

Reply:
Thanks for the explanation @vsoch I was scared by the apparent impossibility to require `build_type=Release` without having a conflict

Reply:
We probably need to change how we handle defaults and the conflicts here - @becker33 whar do you think?